### PR TITLE
authors: use high level categories in research_field

### LINKED ISSUE #2104: author submission: subject autocomplete
My suggestion would be to not list all arXiv subcategories, like math.AG, math.AT... but only higher-level categories. With the exception of physics.ins-det, physics.acc-phys, physics-data.an (these are the ones we take completely in HEP)

### PR DESCRIPTION
**Pull request by chris-asl**
Closes #2104

### CODE REVIEW

File: inspirehep/modules/authors/forms.py [hunk 1]
```diff
@@ -449,6 +449,8 @@ class AuthorUpdateForm(INSPIREForm):
     research_field_options = [
         (val, val)
         for val in arxiv_categories_schema['enum']
+        if '.' not in val or
+           val in ('physics.ins-det', 'physics.acc-ph', 'physics.data-an')
```
Comment (jmartinm) [inline inspirehep/modules/authors/forms.py]: This should be aligned with the line above (not indented)

**Comment by jmartinm** [comment]
To add a test for this code, I have added a comment to a previous PR that had a similar change (https://github.com/inspirehep/inspire-next/pull/2200/files#r113871190) to see what would be a good way of testing it. @jacquerie can you have a look at that?

**Comment by jmartinm** [comment]
Since we agreed that this is ready. @chris-asl just rebase and we can merge it.

**Comment by jacquerie** [comment]
Also please squash and fix the commit message:
```
authors: use high level categories in research_field

* Users requested that only the high level categories and some
  subcategories of physics should be presented as options for
  the `research_field` (closes #2104).

Signed-off-by: Chris Aslanoglou <chris.aslanoglou@gmail.com>
```

**Comment by chris-asl** [comment]
Done! 
Will be using this commit message format henceforth.

**Pull request merged**